File: PushLocal/src/main/java/com/beariksonstudios/automatic/pushlocal/pushlocal/activities/main/NotificationListener.java
package com.beariksonstudios.automatic.pushlocal.pushlocal.activities.main;

import android.content.Intent;
import android.graphics.Bitmap;
import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;
import com.beariksonstudios.automatic.pushlocal.pushlocal.server.Server;

/**
 * Created by nphel on 8/21/2015.
 */
public class NotificationListener extends NotificationListenerService {
    public static final String NOTIFICATION_ACTION = MainActivity.BROADCAST_PREFIX + "Notification";
    public static final String NOTIFICATION_ACTION_NOTIFICATION = "Notification";

    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        String from = sbn.getPackageName();
        String title = sbn.getNotification().extras.getString("android.title");
        String text = sbn.getNotification().extras.getString("android.text");
        String subText = sbn.getNotification().extras.getString("android.subText");
        String notification = from + Server.UNIT + title + Server.UNIT + text + Server.UNIT + subText;
        Log.d("PushLocal", notification);
        Intent intent = new Intent();
        intent.putExtra(NOTIFICATION_ACTION_NOTIFICATION, notification);
        intent.setAction(NOTIFICATION_ACTION);
        sendBroadcast(intent);
    }
}

File: PushLocal/src/main/java/com/beariksonstudios/automatic/pushlocal/pushlocal/activities/main/NotificationListener.kt
package com.beariksonstudios.automatic.pushlocal.pushlocal.activities.main

import android.annotation.TargetApi
import android.content.Intent
import android.graphics.Bitmap
import android.os.Build
import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification
import android.util.Log
import com.beariksonstudios.automatic.pushlocal.pushlocal.server.Server

/**
 * Created by nphel on 8/21/2015.
 */
@TargetApi(Build.VERSION_CODES.KITKAT)
class NotificationListener : NotificationListenerService() {

    override fun onNotificationPosted(sbn: StatusBarNotification) {
        val from = sbn.packageName
        val title = sbn.notification.extras.getString("android.title")
        val text = sbn.notification.extras.getString("android.text")
        val subText = sbn.notification.extras.getString("android.subText")
        val notification = from + Server.UNIT + title + Server.UNIT + text + Server.UNIT + subText
        Log.d("PushLocal", notification)

        val intent = Intent()
        intent.putExtra(NOTIFICATION_ACTION_NOTIFICATION, notification)
        intent.action = NOTIFICATION_ACTION
        sendBroadcast(intent)
    }

    override fun onNotificationRemoved(sbn: StatusBarNotification) {
        super.onNotificationRemoved(sbn)
    }

    companion object {
        val NOTIFICATION_ACTION = MainActivity.BROADCAST_PREFIX + "Notification"
        val NOTIFICATION_ACTION_NOTIFICATION = "Notification"
    }
}
